src/sentry/api/endpoints/project_metrics.py

```diff
@@ -63,14 +79,24 @@ def _handle_by_operation_type(
 
         if metric_operation_type == MetricOperationType.BLOCK_METRIC:
             patched_metrics = block_metric(metric_mri, [project])
+            self._create_audit_log_entry("METRIC_BLOCK", metric_mri, None, project)
         elif metric_operation_type == MetricOperationType.UNBLOCK_METRIC:
             patched_metrics = unblock_metric(metric_mri, [project])
+            self._create_audit_log_entry("METRIC_UNBLOCK", metric_mri, None, project)
         elif metric_operation_type == MetricOperationType.BLOCK_TAGS:
-            tags = request.data.get("tags") or []
+            tags = request.data.get("tags")
+            if not tags:
+                raise InvalidParams("You must supply at least one tag to block")
+
             patched_metrics = block_tags_of_metric(metric_mri, set(tags), [project])
+            self._create_audit_log_entry("METRIC_TAGS_BLOCK", metric_mri, tags, project)
         elif metric_operation_type == MetricOperationType.UNBLOCK_TAGS:
-            tags = request.data.get("tags") or []
+            tags = request.data.get("tags")
+            if not tags:
+                raise InvalidParams("You must supply at least one tag to unblock")
```

Comment:
Added error handling here, since it doesn't make sense to block tags of a metric while supplying no tags.